test/typecheck/struct.c

```diff
@@ -13,5 +13,15 @@ int main() {
     int penny;
   };
 
+  struct birth bd1 = {
+    .date = 1,
+    .month = 1,
+    .year = 1995,
+  };
+
+  struct birth bd2 = {3, 3, 1998};
+
+  struct birth bd3[3] = { [0].date = 4, [1].year = 1999};
```

Comment:
Minor formatting issue; the `.exp` file may have to update accordingly.
```suggestion
  struct birth bd3[3] = {[0].date = 4, [1].year = 1999};
```